src/library/cmd_reader.h

```diff
@@ -184,5 +163,153 @@ namespace xlang::cmd
         }
 
         std::map<std::string_view, std::vector<std::string>> m_options;
+
+        template<typename O, typename L>
+        void extract_option(std::string_view arg, O const& options, L& last)
+        {
+            if (arg[0] == '-')
+            {
+                arg.remove_prefix(1);
+                last = find(options, arg);
+
+                if (last == options.end())
+                {
+                    throw_invalid("Option '-", arg, "' is not supported");
+                }
+
+                m_options.try_emplace(last->name);
+            }
+            else if (arg[0] == '@')
+            {
+                arg.remove_prefix(1);
+                extract_response_file(arg, options);
+            }
+            else if (last == options.end())
+            {
+                throw_invalid("Value '", arg, "' is not supported");
+            }
+            else
+            {
+                m_options[last->name].push_back(std::string{ arg });
+            }
+        }
+
+        template<typename O>
+        void extract_response_file(std::string_view const& arg, O const& options)
+        {
+            std::experimental::filesystem::path response_path{ std::string{ arg } };
+            std::string extension = response_path.extension().generic_string();
+            std::transform(extension.begin(), extension.end(), extension.begin(),
+                [](unsigned char c) { return static_cast<unsigned char>(::tolower(c)); });
+            if (is_directory(response_path) || extension == ".winmd")
```

Comment:
please add comment that this line is intended to help diagnose misuse of the '@' prefix (e.g., when supplying metadata files)